src/System.Net.Http/src/System/Net/Http/Managed/AuthenticationHandler.cs

```diff
@@ -53,17 +49,26 @@ private bool TrySetBasicAuthToken(HttpRequestMessage request)
 
                 foreach (AuthenticationHeaderValue h in authenticateValues)
                 {
-                    // We only support Basic auth, ignore others
-                    if (h.Scheme == Basic)
+                    // We only support Basic and digest auth, ignore others
+                    if (h.Scheme == AuthenticationHelper.Basic)
                     {
-                        if (!TrySetBasicAuthToken(request))
+                        if (AuthenticationHelper.TrySetBasicAuthToken(request, _credentials))
                         {
+                            response.Dispose();
```

Comment:
Should the response body be drained if it's small enough to avoid throwing away a socket from the connection pool? Or does calling dispose on the response already for this?